src/backend/marsha/core/views.py

```diff
@@ -397,6 +399,95 @@ class DocumentView(BaseLTIView):
     serializer_class = DocumentSerializer
 
 
+@method_decorator(csrf_exempt, name="dispatch")
+@method_decorator(xframe_options_exempt, name="dispatch")
+class LTISelectView(TemplateView):
+    """LTI select view."""
+
+    template_name = "core/lti_select.html"
+
+    # pylint: disable=unused-argument
+    def get_request_context_data(self, **kwargs):
+        """Build the context necessary to build lti select form."""
+        lti = LTI(self.request)
+        lti.verify()
+
+        form_data = {"content_items": ""}
+        form_data.update(self.request.POST.items())
+
+        videos = get_resources(Video, lti)
+        public_videos = Video.objects.select_related("playlist").filter(
```

Comment:
This query already exists, you should find a way to factorize what is already existing.